### PR TITLE
Introduce MasterElection stub for testing

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This PR factors out `MasterElection` stub, makes it thread-safely mutable, and refactors tests which used the old stub.

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
Another bit of #1144.

**Comment by pavelkalinnikov** [comment]
PTAL.

File: util/election/stub/election.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+// error if not currently the master.
+func (e *MasterElection) GetCurrentMaster(context.Context) (string, error) {
+	e.mu.RLock()
+	defer e.mu.RUnlock()
+	if e.errs.GetMaster != nil {
+		return "", e.errs.GetMaster
+	}
+	if e.isMaster {
+		return "self", nil
+	}
```
Comment (daviddrysdale) [inline util/election/stub/election.go]: Might be nice if `stub.GetCurrentMaster()` could potentially return a different value (`"someone-else"`), but given that `GetCurrentMaster()` isn't used much at the moment (@AlCutter , is that expected to change?) that's not really needed atm.

Comment (pavelkalinnikov) [reply util/election/stub/election.go]: Yeah, let us add such behavior when it's needed. Leaving as is.

**Pull request merged**